### PR TITLE
[SPARK-31311][SQL][TESTS] Benchmark date-time rebasing in ORC datasource

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to add new benchmarks to `DateTimeRebaseBenchmark` for saving and loading dates/timestamps to/from ORC files. I extracted common code from the benchmark for Parquet datasource and place it to the methods `caseName()` and `getPath()`. Added benchmarks for ORC save/load dates before and after 1582-10-15 because an implementation may have different performance for dates before the Julian calendar cutover day, see #28067 as an example.

### Why are the changes needed?
To have the base line for future optimizations of `fromJavaDate()`/`toJavaDate()` and `toJavaTimestamp()`/`fromJavaTimestamp()` in `DateTimeUtils`. The methods are used while saving/loading dates/timestamps by ORC datasource.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
By running the updated benchmark `DateTimeRebaseBenchmark` via the command:
```
SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "sql/test:runMain org.apache.spark.sql.execution.benchmark.DateTimeRebaseBenchmark"
```
in the environment:

| Item | Description |
| ---- | ----|
| Region | us-west-2 (Oregon) |
| Instance | r3.xlarge |
| AMI | ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-amd64-server-20190722.1 (ami-06f2f779464715dc5) |
| Java | OpenJDK 1.8.0_242-8u242/11.0.6+10 |

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Could you file a JIRA before making a PR? You can make a `DRAFT` PR feature of GitHub for `WIP`.

**Comment by MaxGekk** [comment]
@cloud-fan @dongjoon-hyun @HyukjinKwon Please, take a look at the PR.

**Comment by cloud-fan** [comment]
can you fix the conflicts?

File: sql/core/benchmarks/DateTimeRebaseBenchmark-jdk11-results.txt [hunk 1]
```diff
@@ -6,48 +6,89 @@ OpenJDK 64-Bit Server VM 11.0.6+10-post-Ubuntu-1ubuntu118.04.1 on Linux 4.15.0-1
...
+Save timestamps to ORC:                   Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
+------------------------------------------------------------------------------------------------------------------------
+after 1582, noop                                   3003           3003           0         33.3          30.0       1.0X
+before 1582, noop                                  3012           3012           0         33.2          30.1       1.0X
+after 1582                                        41031          41031           0          2.4         410.3       0.1X
+before 1582                                       44436          44436           0          2.3         444.4       0.1X
+
+OpenJDK 64-Bit Server VM 11.0.6+10-post-Ubuntu-1ubuntu118.04.1 on Linux 4.15.0-1063-aws
+Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz
+Load timestamps from ORC:                 Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
```
Comment (cloud-fan) [inline sql/core/benchmarks/DateTimeRebaseBenchmark-jdk11-results.txt:87]: why the rebasing overhead in read is much smaller than write side?

**Comment by cloud-fan** [comment]
it's just benchmark, no need to wait for jenkins.

Thanks, merging to master/3.0!

**Pull request closed without merging**